src/meta/src/stream/meta.rs

```diff
@@ -305,6 +305,17 @@ where
         }
     }
 
+    /// Used in [`crate::barrier::GlobalBarrierManager`]
+    pub async fn migrate_actors(&self, migrate_map: &HashMap<u32, u32>) -> Result<()> {
+        let map = &mut self.core.write().await.table_fragments;
```

Comment:
should be transactional.

Reply:
pub async fn batch_update_table_fragments(&self, table_fragments: &[TableFragments]) -> Result<()> {
        let map = &mut self.core.write().await.table_fragments;

        let mut transaction = Transaction::default();
        for table_fragment in table_fragments {
            if map.contains_key(&table_fragment.table_id()) {
                    table_fragment.upsert_in_transaction(&mut transaction).await?;
                } else {
                return Err(RwError::from(InternalError(format!(
                    "table_fragment not exist: id={}",
                    table_fragment.table_id()
                ))));
            }
        }
        
        self.meta_store.txn(transaction).await?;
        for table_fragment in table_fragments {
            map.insert(table_fragment.table_id(), table_fragment.clone());
        }
        
        Ok(())
    }
// add param migration_map.